src/service/viewport-impl.js

```diff
@@ -286,8 +286,25 @@ export class Viewport {
    * @return {!../layout-rect.LayoutRectDef}}
    */
   getLayoutRect(el) {
-    return this.binding_.getLayoutRect(el,
-        this.getScrollLeft(), this.getScrollTop());
+    const scrollLeft = this.getScrollLeft();
+    const scrollTop = this.getScrollTop();
+
+    // Go up the window hierarchy through friendly iframes.
+    const win = (el.ownerDocument || el).defaultView;
+    if (win && win != this.win_ && getTopWindow(win) == this.win_) {
```

Comment:
Same logic as https://github.com/ampproject/amphtml/pull/4916/files#diff-b596009fc4aa42dfa7d85c56dccf78eeR125? Can we extract this into a helper?